include/glow/Backends/QueueBackedDeviceManager.h

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_BACKENDS_QUEUEBACKEDDEVICEMANAGER_H
+#define GLOW_BACKENDS_QUEUEBACKEDDEVICEMANAGER_H
+
+#include "glow/Backends/DeviceManager.h"
+#include "glow/Support/ThreadPool.h"
+
+#include <atomic>
+
+namespace glow {
+
+class QueueBackedDeviceManager : public DeviceManager {
+protected:
+  /// Thread which interfaces with the device.
+  ThreadPool workThread_;
+
+  /// Identifier for next run.
+  std::atomic<runtime::RunIdentifierTy> nextIdentifier_{1};
+
+public:
+  QueueBackedDeviceManager(BackendKind backend);
+  virtual ~QueueBackedDeviceManager();
+
+  /// Initialize the device.
+  void init();
+
+  /// Load the provided module into the device, readyCB will be called when
+  /// ready to use
+  void addNetwork(const Module *module, FunctionMap functions,
+                  ReadyCBTy readyCB);
+
+  /// Remove (and delete) the provided network and all it's functions, freeing
+  /// up space on the device.
+  void evictNetwork(const Module *module);
```

Comment:
We should keep it consistent with addNetwork and the ReadyCB, and since those are used to do comparisons I think better to keep it as pointer (ie. do a ptr compare, not an operator==).

Honestly I think the raw ptrs here should be shared_ptrs.